service/history/workflow/mutable_state_impl.go

```diff
@@ -572,6 +573,42 @@ func (e *MutableStateImpl) GetLastWriteVersion() (int64, error) {
 	return common.EmptyVersion, nil
 }
 
+// GetDynamicAttributes returns the memo and search attributes of the workflow execution.
+// If the close execution visibility task is incomplete, this method fetches those fields from the mutable state.
+// Otherwise, it fetches the memo from the visibility storage backend.
+func (e *MutableStateImpl) GetDynamicAttributes(ctx context.Context, visibilityManager manager.VisibilityManager) (*DynamicAttributes, error) {
```

Comment:
Switched back to the old version. I'm not implementing the cache for now because I don't want to affect the existing implementation. However, I did leave a comment and a TODO For it.